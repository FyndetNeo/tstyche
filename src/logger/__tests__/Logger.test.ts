import { describe, expect, jest, test } from "@jest/globals";
import ansiEscapesSerializer from "jest-serializer-ansi-escapes";

expect.addSnapshotSerializer(ansiEscapesSerializer);

const mockedTerminalStream = {
  // eslint-disable-next-line @typescript-eslint/naming-convention
  isTTY: true,
  write: jest.fn(),
} as unknown as NodeJS.WriteStream;

const mockedNonTerminalStream = {
  write: jest.fn(),
} as unknown as NodeJS.WriteStream;

const { Logger } = await import("../Logger.js");

describe("Logger", () => {
  describe("'eraseLastLine' method", () => {
    test.each([
      {
        expected: true,
        stdout: mockedTerminalStream,
        testCase: "when the 'stdout' stream is interactive",
      },
      {
        expected: false,
        stdout: mockedNonTerminalStream,
        testCase: "when the 'stdout' stream is not interactive",
      },
    ])("$testCase", ({ stdout }) => {
      const logger = new Logger({ stdout });

      logger.eraseLastLine();

      expect(jest.mocked(mockedTerminalStream.write).mock.calls[0]).toMatchSnapshot();
    });
  });

  describe("'isInteractive' method", () => {
    test.each([
      {
        expected: true,
        stdout: mockedTerminalStream,
        testCase: "when the 'stdout' stream is interactive",
      },
      {
        expected: false,
        stdout: mockedNonTerminalStream,
        testCase: "when the 'stdout' stream is not interactive",
      },
    ])("$testCase", ({ expected, stdout }) => {
      const logger = new Logger({ stdout });

      expect(logger.isInteractive()).toBe(expected);
    });
  });
});
