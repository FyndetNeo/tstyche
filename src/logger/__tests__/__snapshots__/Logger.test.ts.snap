// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logger 'eraseLastLine' method when the 'stdout' stream is interactive 1`] = `
[
  "<moveCursorUpBy1Row>
<eraseLineEnd>
",
]
`;

exports[`Logger 'eraseLastLine' method when the 'stdout' stream is not interactive 1`] = `undefined`;
